test/table.js

```diff
@@ -1378,6 +1378,248 @@ describe('Bigtable/Table', function() {
+        assert.strictEqual(token, response.consistency_token);
+        callback(null, {consistent: true});
+      };
+
+      table.waitForReplication(() => {
+        done();
+      });
+    });
+
+    describe('retries', () => {
+      let clock;
+      let setTimeoutSpy;
+      let clearTimeoutSpy;
+      let checkConsistencySpy;
+      let responses = [];
+
+      beforeEach(() => {
+        clock = sinon.useFakeTimers({
+          toFake: ['setTimeout', 'clearTimeout'],
+        });
+        setTimeoutSpy = sinon.spy(global, 'setTimeout');
+        clearTimeoutSpy = sinon.spy(global, 'clearTimeout');
+        checkConsistencySpy = sinon.spy(table, 'checkConsistency');
+
+        table.bigtable.request = function(config, callback) {
+          responses.shift()(config, callback);
+        };
+      });
+
+      afterEach(() => {
+        clock.restore();
+      });
+
+      it('should return true if token is consistent', done => {
+        responses = [
+          (config, callback) =>
+            callback(null, {consistency_token: 'sample-token12345'}),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          assert.strictEqual(checkConsistencySpy.callCount, 1);
+
+          // Checks that clearInterval was called.
+          assert.strictEqual(clearTimeoutSpy.callCount, 1);
+
+          assert.strictEqual(response, true);
+          assert.ifError(err);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should retry checkConsistency', done => {
+        responses = [
+          (config, callback) =>
+            callback(null, {consistency_token: 'sample-token12345'}),
+          (config, callback) => callback(null, {consistent: false}),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          checkConsistencySpy.callOnce;
+
+          setTimeoutSpy.calledWith(sinon.match.func, 5000);
+
+          // check checkConsistencySpy called twice after 5seconds
+          clock.tick(5010);
+          assert.strictEqual(checkConsistencySpy.callCount, 2);
+
+          // Checks that clearInterval was called.
+          setTimeoutSpy.called;
+          assert.ifError(err);
+          assert.strictEqual(response, true);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return false after 10 min if inconsistency repeats', done => {
+        this.timeout(10 * 61 * 1000); // 10 minutes
```

Comment:
All the `this.timeout` calls should be unnecessary and removed, they should be throwing errors since they're in fat arrow functions without a higher `this` scoped 🤷‍♂️

Reply:
this.timeout removed https://github.com/googleapis/nodejs-bigtable/pull/82/commits/5ba4e36e0e7b47a136f301cf7d9a01473e68add6